.github/workflows/ci.yml

```diff
@@ -42,6 +42,16 @@ jobs:
       - run: ./ci/test.sh
         env:
           WORKDIR: ${{ github.workspace }}
+  go-mod-tidy-check:
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@c85c95e3d7251135ab7dc9ce3241c5835cc595a9 # v3.5.3
+      - uses: actions/setup-go@fac708d6674e30b6ba41289acaab6d4b75aa0753 # v4.0.1
+        with:
+          go-version: '1.22.2'
```

Comment:
Maybe this could be used? https://github.com/actions/setup-go?tab=readme-ov-file#getting-go-version-from-the-gomod-file